localstack-core/localstack/services/events/event_rule_engine.py

```diff
@@ -109,15 +109,15 @@ def _evaluate_condition(self, value, condition, field_exists: bool):
             else:
                 return self._evaluate_suffix(suffix, value)
 
-        elif equal_ignore_case := condition.get("equals-ignore-case"):
+        elif (equal_ignore_case := condition.get("equals-ignore-case")) is not None:
             return self._evaluate_equal_ignore_case(equal_ignore_case, value)
         elif numeric_condition := condition.get("numeric"):
```

Comment:
The isinstance check should handle the numeric case, hence we don't need the None check here

Reply:
added a comment 👍